src/main/kotlin/lotto/domain/LottoNumber.kt

```diff
@@ -0,0 +1,22 @@
+package lotto.domain
+
+@JvmInline
+value class LottoNumber(val number: Int) {
+    companion object {
+        private const val MINIMUM_NUMBER = 1
+        private const val MAXIMUM_NUMBER = 45
+        private const val CANNOT_CREATE_LOTTO_NUMBER_MESSAGE = "Lotto number must be between %s and %s."
+
+        private val NUMBERS: Map<Int, LottoNumber> = (MINIMUM_NUMBER..MAXIMUM_NUMBER).associateWith { LottoNumber(it) }
+
+        fun from(number: Int): LottoNumber {
+            return NUMBERS[number] ?: throw IllegalArgumentException(
+                CANNOT_CREATE_LOTTO_NUMBER_MESSAGE.format(MINIMUM_NUMBER, MAXIMUM_NUMBER),
+            )
+        }
+    }
```

Comment:
```suggestion
@JvmInline
value class LottoNumber(val number: Int) {

    init {
        require(NUMBERS[number] != null) {
            CANNOT_CREATE_LOTTO_NUMBER_MESSAGE.format(MINIMUM_NUMBER, MAXIMUM_NUMBER)
        }
    }

    companion object {
        private const val MINIMUM_NUMBER = 1
        private const val MAXIMUM_NUMBER = 45
        private const val CANNOT_CREATE_LOTTO_NUMBER_MESSAGE = "Lotto number must be between %s and %s."
        private val NUMBERS: Map<Int, LottoNumber> = (MINIMUM_NUMBER..MAXIMUM_NUMBER).associateWith { LottoNumber(it) }
    }
    ...
```

저는 코틀린을 사용하면서 정적 팩토리 메서드는 좀 더 명확하게 쓰이게 되더라구요. 기본 프로퍼티인 경우에는 기본 생성자를 쓰게 되고, 그 외 기본 프로퍼티와 다른 타입의 프로퍼티가 있으면 정적 팩토리 메서드를 사용하는 게 더 어울린다고 생각합니다. 그럼으로써 기본 프로퍼티를 적극적으로 사용하면, 초기 검증 로직도 init에 넣을 수 있어 더 직관적으로 보이는 부분도 있었어요.

이처럼 기본 생성자를 적극적으로 사용하고, init을 적극적으로 사용해보는 건 어떨까요?